src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -3159,10 +3200,45 @@ DeformBackgroundTaskHeapTuple(TupleDesc tupleDescriptor, HeapTuple taskTuple)
 			TextDatumGetCString(values[Anum_pg_dist_background_task_message - 1]);
 	}
 
+	if (!nulls[Anum_pg_dist_background_task_node_tokens - 1])
+	{
+		task->nodeTokens =
+			datumIntArrayToList(values[Anum_pg_dist_background_task_node_tokens - 1]);
+	}
+
 	return task;
 }
 
 
+/*
+ * datumIntArrayToList
+ * See PG's oid_array_to_list implementation for reference
+ */
+static List *
+datumIntArrayToList(Datum datum)
+{
+	ArrayType *array = DatumGetArrayTypeP(datum);
```

Comment:
Apart from this call, I think you can use `IntegerArrayTypeToList` to do all the deconstruction.

Reply:
Good point
(😭 wish I knew that sooner)